setup.py

```diff
@@ -13,8 +13,8 @@
 
 if sys.version_info < py_min_version:
     error = """
-Beginning with Matplotlib {0}, Python {1} or above is required.
-You are using Python {2}.
```

Comment:
This should stay to error out correctly on Python 2.6, though I hope no-one is attempting to run Matplotlib 3.8 there.

Reply:
You are far more kind than I am! If someone tries to run this with Python 2.6 I would be fine with them failing in other ways :)
I did revert though.

Reply:
If this is a priority, probably _all_ of the f-strings in setup.py need to be reverted, otherwise it will SyntaxError before it runs, won't it? (been a while since I ran _anything_ in py2, but certainly syntax errors in general are raised at parse time, not at line execution time)

Reply:
Indeed:
```
  File "setup.py", line 114
    f'{name} contains -fno-lto')
    ^
SyntaxError: invalid syntax
```

I would definitely prefer to just leave it as-is, but I'm fine reverting this entire file as well if you prefer that.

Reply:
I guess in my opinion one of the two must happen (I don't have _strong_ opinions on which, but a relatively strong opinion on staying consistent):

- The comment is edited/deleted to remove the assertion that the file is "compatible with python 2"

or

- The f-strings are removed to make the comment correct

Python 2 was officially EOL 2020-01-01, so I don't feel _too_ bad about not having a more informative message for py2. Though the difference between `pip install ...` and `pip3 install ...` on systems which have py2 may be a compelling reason: somebody sees "we require python 3.9+" and they go "OHHH, oops. haha", but they see "SyntaxError" and go "It must be the setup.py that is wrong!".
We could say something similar about py <=3.5, but those are also EOL as of 2020-09-30, so being 2-3 years beyond the EOL dates, not _too_ concerned.

Reply:
OK, I reverted the `setup.py` file.

Reply:
Ah, I forgot about the `SyntaxError`, tbh, but I think once we move the extensions all to pybind11 and can move more to `pyproject.toml`, that can be a good time to make a clean break of everything, since `setup.py` will hopefully be gone (or more empty, at least).